### PR TITLE
DNN-16884: Fix Page Parent style when parent page name has no spaces …

### PR DESCRIPTION
**Pull request by ricardodiasc**
…and it is too big

### CODE REVIEW

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx [hunk 1]
```diff
@@ -33,6 +33,32 @@ class PageTypeSelector extends Component {
         const hierarchy = this._getHierarchyLabel();        
         const components = this.getComponents(); 
 
+        let pageParentInfoStyle = null;
+        let parentPageStyleLabel = null;
+        let parentPageStyleContent = null;
+        
+
+        if (hierarchy.length < 31) {
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx]: Why did you choose 31 as the cutoff length? Keeping in mind that the font is variable width, is it low enough to give a desired effect when the letters are wider on average?

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx [hunk 2]
```diff
@@ -51,13 +77,13 @@ class PageTypeSelector extends Component {
                                 {createdDate}
                             </span>
                         </div>
-                        <div className="page-info-item">
-                            <span className="page-info-item-label">
+                        <div className="page-info-item" style={pageParentInfoStyle}>
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx]: I'd rather we avoid inline styles and instead use class or attribute selectors. Is there some problem with selector specificity which blocks us?

What I have in mind is to set a class based on the logic above

```
<div className="page-info-item {hierarchy.length > 31 ? page-info-long : null}" />
```

And style those divs purely in css. Thoughts @emac80 ?

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx (hunk 1 above)
Comment (ricardodiasc) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx]: Agree on that. I will change to find out the width if this field

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx [hunk 3]
```diff
@@ -24,41 +30,33 @@ class PageTypeSelector extends Component {
...
     }
+
+    componentDidMount() {
+        const pageSelectorDOM = ReactDOM.findDOMNode(this);
+        if (pageSelectorDOM.querySelector("#parentPageValue").offsetWidth > 180) {
+            pageSelectorDOM.querySelector("#pageParent").className = "page-info-item page-parent-info-style-large";
+            pageSelectorDOM.querySelector("#pageParentLabel").className = "page-info-item-label parent-page-style-label-large";
+            pageSelectorDOM.querySelector("#pageParentContent").className = "page-info-item-value parent-page-name parent-page-style-content-large";
+        }
+        console.log("Component did mount",pageSelectorDOM.querySelector("#parentPageValue").offsetWidth);
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx]: Looks like a debugging log

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx [hunk 4]
```diff
@@ -24,41 +30,33 @@ class PageTypeSelector extends Component {
             return page.hierarchy;
         }
     }
+
+    componentDidMount() {
+        const pageSelectorDOM = ReactDOM.findDOMNode(this);
+        if (pageSelectorDOM.querySelector("#parentPageValue").offsetWidth > 180) {
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageTypeSelector/PageTypeSelector.jsx]: Let's have this magical 180 extracted into a constant and we'll be done here!

**Review by tpluscode — APPROVED** [review]
Nicely done. Using actual element dimensions is so much better! 💖 

Please address the last two nitpicks and I'll be happy to merge.

**Pull request merged**